src/core/shape_inference/include/reverse_sequence_shape_inference.hpp

```diff
@@ -26,25 +28,33 @@ void shape_infer(const ReverseSequence* op, const std::vector<T>& input_shapes,
                           seq_lengths_rank.compatible(1),
                           "Sequence lengths rank must be equal to 1. Got: ",
                           seq_lengths_pshape);
-    auto& output_pshape = output_shapes[0];
-
-    output_pshape = data_pshape;
+    auto output_pshape = data_pshape;
     if (data_rank.is_static() && seq_lengths_rank.is_static()) {
+        const auto normalized_batch_axis = ov::normalize_axis(op, op->m_batch_axis, data_rank);
```

Comment:
let's call `get_batch_axis` here and get rid of this friend function in the ReverseSequence class. It just makes it more complex with no obvious reason why.

Reply:
Good catch, `get_origin_batch_axis()` method as it give `m_batch_axis`. The `get_batch_axis ` can not work as it use input 0 for rank value which can be not set when operator created with default ctor.